pkg/catalog/routes.go

```diff
@@ -237,28 +237,30 @@ func getTrafficPolicyPerRoute(mc *MeshCatalog, routePolicies map[trafficpolicy.T
 			Namespace: trafficTargets.Spec.Destination.Namespace,
 			Name:      trafficTargets.Spec.Destination.Name,
 		}
-		destService, destErr := mc.GetServiceForServiceAccount(dstNamespacedServiceAcc)
+		destServiceList, destErr := mc.GetServicesForServiceAccount(dstNamespacedServiceAcc)
 		if destErr != nil {
 			log.Error().Msgf("TrafficTarget %s/%s could not get destination services for service account %s", trafficTargets.Namespace, trafficTargets.Name, dstNamespacedServiceAcc.String())
 			return nil, destErr
 		}
+		destService := destServiceList[0]
```

Comment:
would be good to create an issue and reference it wherever these assumptions are being made.

Reply:
Updated issue #1550